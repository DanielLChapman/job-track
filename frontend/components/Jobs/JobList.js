import React from 'react';
import PropTypes from 'prop-types';
import JobView from './JobView';

JobList.propTypes = {
    jobs: PropTypes.arrayOf(PropTypes.object),
};

function JobList(props) {
    return (
        <ul className="job-view-container">   
            {
                props.jobs && props.jobs.map((x) => (
                    <li key={x.id}>
                        <JobView job={x} modalFill={props.modalFill} />
                    </li>
                ))
            }
        </ul>
    );
}

export default JobList;