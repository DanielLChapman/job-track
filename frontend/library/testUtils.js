import casual from 'casual';

casual.seed(231313);

const fakeUser = (overrides) => ({
    __typename: 'User',
    id: '4234',
    name: casual.name,
    jobs: [],
    email: casual.email,
    ...overrides,
  });

  const fakeJob = (overrides) => ({
    __typename: 'User',
    id: '4234',
    name: casual.name,
    publishedAt: casual.date('YYYY-MM-DD'),
    applicationDate: casual.date('YYYY-MM-DD'),
    status: "waiting",
    author: fakeUser().id,
    salaryExpectation: 0,
    notes: casual.sentence,
    ...overrides,
  });

  export {
    fakeUser,
    fakeJob
  }
