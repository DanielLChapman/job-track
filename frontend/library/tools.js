export const convertNum = (num) => {
    let t = num.toString();
    if (t.length < 3) {
        t = "$" + t
        return t;
    }
    let result = [];
    let count = 1;
    for (let i = t.length - 1; i >= 0; i--) {
        result.unshift(t[i]);
        
        if (count % 3 === 0 && i !== 0) {
            result.unshift(',')
        }
        count++;
    }

    result.unshift('$')

    return result.join('');

}