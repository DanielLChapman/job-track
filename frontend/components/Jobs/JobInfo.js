import React from 'react';
import PropTypes from 'prop-types';
import { convertNum } from '../../library/tools';

JobInfo.propTypes = {
    job: PropTypes.object.isRequired,
};

function JobInfo({job}) {
    return (
        <ul>
            <li><h2>{job.name}</h2></li>
            <li><h5>{job.status}</h5></li>
            <li><h5>{convertNum(job.salaryExpectation)}</h5></li>
            <li><h5>{job.notes}</h5></li>
        </ul>
    );
}

export default JobInfo;