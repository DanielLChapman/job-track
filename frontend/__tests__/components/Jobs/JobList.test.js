import {act, screen, render} from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import wait from 'waait';
import { fakeJob } from '../../../library/testUtils';
import JobInfo from '../../../components/Jobs/JobInfo';
import JobList from '../../../components/Jobs/JobList';

const jobs = [{...fakeJob(), id: 1234}, {...fakeJob(), id: 1244}, {...fakeJob(), id: 12234}];

describe('It insures the correct information is displayed', () => {
    it ('renders correctly when jobs are passed in', async () => {

        const { container, debug} = render(
            <MockedProvider>
                <JobList jobs={jobs} />
            </MockedProvider>
        )

        await act(async() => {
            await wait(0);
        });

        await screen.findByText(jobs[0].name);
        await screen.findByText(jobs[1].name);
        await screen.findByText(jobs[2].name);

        expect(container).toMatchSnapshot();
    })
})