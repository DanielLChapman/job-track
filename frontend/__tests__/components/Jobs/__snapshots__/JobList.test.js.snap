// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It insures the correct information is displayed renders correctly when jobs are passed in 1`] = `
<div>
  <ul
    class="job-view-container"
  >
    <li>
      <section
        class="job-block"
        data-testid="job-block"
      >
        <section
          class="job-block-title"
        >
          <h5
            class="job-block-title-text"
          >
            Mrs. Constantin Dickinson
          </h5>
          <span
            class="job-block-status"
          >
            waiting
          </span>
          <section
            class="button-group"
          >
            <button
              class="job-block-title-delete"
              type="button"
            >
              Delete
            </button>
            <button
              type="button"
            >
              ⇩
            </button>
          </section>
        </section>
      </section>
    </li>
    <li>
      <section
        class="job-block"
        data-testid="job-block"
      >
        <section
          class="job-block-title"
        >
          <h5
            class="job-block-title-text"
          >
            Miss Amira Kuvalis
          </h5>
          <span
            class="job-block-status"
          >
            waiting
          </span>
          <section
            class="button-group"
          >
            <button
              class="job-block-title-delete"
              type="button"
            >
              Delete
            </button>
            <button
              type="button"
            >
              ⇩
            </button>
          </section>
        </section>
      </section>
    </li>
    <li>
      <section
        class="job-block"
        data-testid="job-block"
      >
        <section
          class="job-block-title"
        >
          <h5
            class="job-block-title-text"
          >
            Miss Terry Schaefer
          </h5>
          <span
            class="job-block-status"
          >
            waiting
          </span>
          <section
            class="button-group"
          >
            <button
              class="job-block-title-delete"
              type="button"
            >
              Delete
            </button>
            <button
              type="button"
            >
              ⇩
            </button>
          </section>
        </section>
      </section>
    </li>
  </ul>
</div>
`;
